docs/adr/0002-project-locks.md

```diff
@@ -0,0 +1,160 @@
+
+This is due to a combination of different directories passed to Working Dir when cloning the repository during the pre-workflow hook and during a plan.
+
+ ## Goals & Non-Goals
+ 
+ ### Goals
+ - Alignment on what Atlantis should or shouldn't be locking
+ - Focus on small scope changes related only to locking
+ - Supports all workflow use-cases 
+	 - Terraform w/o workspaces
+	 - Terraform w/ workspaces
+	 - Terragrunt monorepo
+ 
+### Non-Goals
+- Making changes/improvements to other packages/sub-systems
+- Avoid massive refactor if possible
+- Focus on a singular workflow
+
+## Solutions
+
+### Previous attempts
+ There have been a couple of PRs submitted that have either been reverted, or grown stale:
+
+- add path to WorkingDir [#2180](https://github.com/runatlantis/atlantis/pull/2180)
+	- base32 encodes the path provided
+	- Uses the unique base32 string to additional clone the repo to unique directories to avoid plan file overlap per project + workspace
+- add non-default workspaces to workflow hooks [#2882](https://github.com/runatlantis/atlantis/pull/2882)
+    - reintroduces #2180 changes to working_dir
+    - run hooks on the default workspace
+    - also run hooks on every project found
+    - significant increase in execution time, especially for Terragrunt users relying on terragrunt-atlantis-config
+- reduce the amount of git clone operations [#2921](https://github.com/runatlantis/atlantis/pull/2921)
+    - started to try to reduce execution time on #2882 implementation
+    - attempts to utilize [TF_DATA_DIR](https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_data_dir) for workspaces and remove the workspace for the lock. 
+    - Clones only at BaseRepo/PR# and moves workspaces from Atlantis clones to TF workspace managed
+
+This is not to suggest the revival of these PRs in their current state but to act as a reference for additional focused solutions. 
+
+### Possible Solutions
+**Note**: Some of these solutions will purposely be contradictory to the ADR Goals or Non-Goals in an effort to flesh out alternatives and to assist in the decision-making process.
+
+
+
+### Solution 1: Clone once + TF_DATA_DIR
+
+Take PR #2921 and re-implement locks on the DATA_DIR only for plans and the entire base repo + PR for git clones.
+
+**TO-DO**: flesh out
+
+### Solution 2: Clone once + move plans into state/backend
+
+Take PR #2921 and remove the concept of locking on workspace/path. Clone only once during Pull Request events (commit updates) and not during Atlantis commands (plan/apply).
+
+**TO-DO**: figure out how to process plan files/and where to output other than disk.
+
+### Solution 3: Revert #2131
+
+The original change that introduced `path` to the WorkingDirLocker could be reverted and fall back on known behavior.
+
+...
+
+## Decision
+
+TBD
+
+## Links
+
+https://github.com/runatlantis/atlantis/issues/1914
```

Comment:
Id bullet these so GitHub automatically grabs each issue title in the markdown preview

E.g.

```suggestion
- https://github.com/runatlantis/atlantis/issues/1914
```